chore: Update influxdb converter to v0.2.16

jacobmarble opened the pull request:
**Description:** <Describe what has changed.>

The [InfluxDB exporter follows](https://github.com/influxdata/influxdb-observability/blob/14db1a7807e294fd3fea47c084818cdb42f07710/otel2influx/metrics_telegraf_prometheus_v2.go#L186-L189) the [OTLP protocol description](https://github.com/open-telemetry/opentelemetry-proto/blob/1648cf3ff58af20e7721c0e95b226c4f3336dee9/opentelemetry/proto/metrics/v1/metrics.proto#L464-L471):

```
  // The number of elements in bucket_counts array must be by one greater than
  // the number of elements in explicit_bounds array.
```

The OTel [Metrics Data Model doc](https://github.com/open-telemetry/opentelemetry-specification/blob/660826e60a9aebad8f97e52e3cb352877c8d8ec7/specification/metrics/datamodel.md#histogram) doesn't add a lot:

>  - (optional) A series of buckets with:
>    - Explicit boundary values.  These values denote the lower and upper bounds
>      for buckets and whether not a given observation would be recorded in this
>      bucket.
>    - A count of the number of observations that fell within this bucket.

However:
- it isn't the exporter's job to enforce convention
- many exporters appear to accept the missing "infinity bucket"
- the Prometheus exporter doesn't even check for the infinity bucket
- the two schemas in the current InfluxDB exporter follows Prometheus convention

Therefore, the InfluxDB exporter should not needlessly reject a histogram that doesn't specify an infinity bucket.

**Link to tracking Issue:** <Issue number if applicable>

Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7049

**Testing:** <Describe what testing was performed and which tests were added.>

Added unit tests to the updated package in https://github.com/influxdata/influxdb-observability/pull/52

**Documentation:** <Describe the documentation added.>
n/a

Commit:
fix: InfluxDB: accept histogram metric missing infinity bucket

Commit:
chore: add changelog entry

anuraaga reviewed (commented):
Is the change completely in the external influxdb-observability repo? In that case, I would make the PR title just "Update influxdb converter to v0.2.16" or something since it's unusual to have a bugfix without any code. The actual fix could still be mentioned in the PR description.

Or if not the case, I can't find the business logic changes in this PR

pmm-sumo commented:
> Is the change completely in the external influxdb-observability repo? In that case, I would make the PR title just "Update influxdb converter to v0.2.16" or something since it's unusual to have a bugfix without any code. The actual fix could still be mentioned in the PR description.
> 
> Or if not the case, I can't find the business logic changes in this PR

That's a good question. We historically included description of changes in the bumped libraries such as opentelemetry-log-collection ([example](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8124/files)), which is used for `filelog` receiver. The caveat was that those changes were in libraries directly used in the collector (almost like components) and in fact implemented with collector in mind. I think this is also the case here. If I would be using InfluxDB that information would be helpful (since changelog is one of the first resources when I'm searching if a possible problem has been already solved with a newer version)

pmm-sumo reviewed (approved):
LGTM overall, there are several `go.sum` though which I think should be not part of this PR

pmm-sumo commented on internal/coreinternal/go.mod:
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/stretchr/testify v1.7.0
 	go.opentelemetry.io/collector v0.46.1-0.20220307173244-f980c9ef25b1
 	go.opentelemetry.io/collector/model v0.46.1-0.20220307173244-f980c9ef25b1
+	go.uber.org/zap v1.21.0
```
Why this (minor) change? (and similar once in attributesprocessor)

jacobmarble replied:
Fixed.

jacobmarble commented:
> If I would be using InfluxDB that information would be helpful (since changelog is one of the first resources when I'm searching if a possible problem has been already solved with a newer version)

Not sure we have consensus. I've updated the PR description, but left the changelog entry as-is.

> there are several go.sum though which I think should be not part of this PR

This is the result of `make gotidy && go mod tidy`. I've reverted that and rebased on a recent checkout of main.

codeboten commented:
Closing and reopening to trigger CI
